vars/washeryCleanSnapshots.groovy

```diff
@@ -0,0 +1,105 @@
+/***********************************
+washeryCleanSnapshots DSL
+
+Cleans old washery snapshots from RDS Snapshot storage
+
+example usage
+executeChangeSet(
+  region: 'us-east-1', // (required, aws region to deploy the stack)
+  accountId: env.DEV_ACCOUNT_ID,
+  role: env.CIINABOXV2_ROLE, // IAM role to assume
+  snapshotRetainCount: 3 // The number of washery snapshots to keep stored
+)
+************************************/
+
+import com.base2.ciinabox.aws.AwsClientBuilder
+import com.amazonaws.services.rds.model.DescribeDBSnapshotsRequest
+import com.amazonaws.services.rds.model.DeleteDBSnapshotRequest
+import com.amazonaws.services.rds.model.DescribeDBClusterSnapshotsRequest
+import com.amazonaws.services.rds.model.DeleteDBClusterSnapshotRequest
+
+def call(body) {
+    def config = body
+    def clientBuilder = new AwsClientBuilder([
+        region: config.region,
+        awsAccountId: config.accountId,
+        role: config.role
+    ])  
+
+    def client = clientBuilder.rds()
+
+    cleanInstanceWasherySnapshots(client, config.snapshotRetainCount)
+    cleanClusterWasherySnapshots(client, config.snapshotRetainCount)
+}
+
+@NonCPS
+def cleanInstanceWasherySnapshots(client, snapshotRetainCount){
+    //RDS Instance
+    def request = new DescribeDBSnapshotsRequest()
+            .withSnapshotType("manual")
+    def snapshotsResult = client.describeDBSnapshots(request)
+    def snapshots = snapshotsResult.getDBSnapshots()
+    def washerySnapshots = []
+
+    //Retrieve snapshots prefixed with `washery-scrubbed`
+    if(snapshots.size() > 0) {
+        for (int i = 0; i < snapshots.size(); i++) {
+            if (snapshots[i].getDBSnapshotIdentifier().startsWith("washery-scrubbed-")){
+                washerySnapshots.add(snapshots[i])
+            }
+        }
+        //Sort washery snapshots based on snapshot create time
+        washerySnapshots.sort{a,b-> a.getSnapshotCreateTime().compareTo(b.getSnapshotCreateTime())}
+        for (int i = 0; i < washerySnapshots.size(); i++) {
```

Comment:
instead of this you could do convert the snapshot list into 2 lists: keep and delete

You could then print them out in 2 separate tables clearly defining what is being deleted and kept. You could even provide an option to manually approve the delete if the customer wanted to

```groovy
def retain = 3
def snapshots = [1,2,3,4,5,6,7,8,9]

def keep = snapshots[0..retain-1]
def delete = snapshots[retain..-1]

println keep
println delete
```

Reply:
Added delete array (Also added condition so we don't get an outofbounds exception when retain count is larger than the size of total snapshots)
```
 //If retain count is less than the size of the total washery snapshots
    if (snapshotRetainCount < washerySnapshots.size()){
        def delete = washerySnapshots[snapshotRetainCount..-1]
        println delete
```